pkg/gitops_test.go

```diff
@@ -1023,16 +1017,10 @@ func TestRemoveAndPush(t *testing.T) {
 	componentBasePath := "/fake/path/test-component/components/test-component/base"
 	componentName := "test-component"
 	component := gitopsv1alpha1.Component{
-		Spec: gitopsv1alpha1.ComponentSpec{
-			Source: gitopsv1alpha1.ComponentSource{
-				ComponentSourceUnion: gitopsv1alpha1.ComponentSourceUnion{
-					GitSource: &gitopsv1alpha1.GitSource{
-						URL: repo,
-					},
-				},
-			},
-			TargetPort: 5000,
+		Source: &gitopsv1alpha1.ComponentSource{
+			URL: repo,
```

Comment:
Add coverage for ComponentSource optional fields

Reply:
Removing those unused fields.